EDU-16348 Using data attributes

mariana-caetano opened the pull request:
#### Types of changes
- [x] New content (guides, endpoints, app documentation)
- [ ] Improvement (make a documentation even better)
- [ ] Fix (fix a documentation error)
- [ ] Spelling and grammar accuracy (self-explanatory)

mariana-caetano committed:
First content without review

mariana-caetano committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

mariana-caetano committed:
Merge branch 'main' into data-attributes

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
````diff
@@ -0,0 +1,533 @@
+```scss
+// Target all Button components
+[data-fs-button] {
+  border-radius: 8px;
````
@mariana-caetano I think we should avoid using hard-coded values here, even if it's just an example, we can use our tokens instead. Consider this adjustment across the entire file.

mariana-caetano replied:
Good point! I've updated all the examples in this doc to use our design tokens.

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
````diff
@@ -0,0 +1,533 @@
+}
+```
+
+### 4. Nested Elements
````
Maybe we can show ProductCard's structure here for some context:

````typescript
<ProductCard>
  <ProductCardImage>
    <img
      data-fs-image
      src={product.image[1].url}
      alt={product.image[1].alternateName}
    />
  </ProductCardImage>
  <ProductCardContent
    title={product.isVariantOf.name}
    price={{
      value: product.offers.offers[0].price,
      listPrice: product.offers.offers[0].listPrice,
      formatter: useFormattedPrice,
    }}
    ratingValue={3.5}
    onButtonClick={() => {}}
  />
</ProductCard>

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
````diff
@@ -0,0 +1,533 @@
+```scss
+// Target only primary small buttons
+[data-fs-button-variant="primary"][data-fs-button-size="small"] {
+  background-color: #0056b3;
+  padding: 6px 12px;
+}
+
+// Target product cards with bordered variant
+[data-fs-product-card][data-fs-product-card-bordered="true"] {
+  border: 1px solid #e0e0e0;
+  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
+}
+```
+
+### 2. Responsive Design
+
+Use data attributes with media queries for responsive styling:
+
+```scss
+// Mobile-first approach
+[data-fs-product-card] {
+  width: 100%;
+  margin-bottom: 16px;
+}
+
+@media (min-width: 768px) {
+  [data-fs-product-card] {
+    width: calc(50% - 8px);
+    display: inline-block;
+  }
+}
+
+@media (min-width: 1024px) {
+  [data-fs-product-card] {
+    width: calc(33.333% - 12px);
+  }
+}
````
```suggestion
// Mobile-first approach
[data-fs-product-card] {
  width: 100%;
  margin-bottom: 16px;
  
  @media (min-width: 768px) {
    width: calc(50% - 8px);
    display: inline-block;
  }
  
  @media (min-width: 1024px) {
    width: calc(33.333% - 12px);
  }
}
```

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
```diff
@@ -0,0 +1,533 @@
+
+### 3. Theme Customization
+
+Create theme variations using data attributes:
```
Here we should reinforce that the proposed way of editing a theme is by subscribing the tokens (the global and component specific ones). Example:

````scss
[data-fs-product-card] {
  --fs-product-card-border-color: transparent;
  --fs-product-card-border-color-hover: var(--fs-border-color-light);

  &[data-fs-product-card-bordered='true'] {
    --fs-product-card-border-color: var(--fs-border-color-light);
  }
}

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
```diff
@@ -0,0 +1,533 @@
+
+### ❌ Don'ts
+
+- **Don't modify core data attributes** - Avoid changing the built-in `data-fs-` attributes
```
I didn't understand this one

mariana-caetano replied:
This one was more to reinforce the use of semantic data attributes and the design tokens we already have built in, but the sentence is indeed confusing. Since I've rewritten [the section for best practices](https://github.com/vtexdocs/dev-portal-content/pull/2256/changes#diff-0ba3d7006eff78afb84baefbbb319c2aecc85ab6546902e4633efb1f86fa2dbaR390) to reinforce these points with examples, I removed this sentence.

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
````diff
@@ -0,0 +1,533 @@
+    transform: translateY(-2px);
+    box-shadow: 0 6px 20px rgba(102, 126, 234, 0.6);
+  }
+}
+```
+
+### Custom Product Card Layout
+
+```scss
+// Custom product card with hover effects
+[data-fs-product-card] {
+  transition: all 0.3s ease;
+  border-radius: 12px;
+  overflow: hidden;
+  
+  &:hover {
+    transform: translateY(-4px);
+    box-shadow: 0 8px 25px rgba(0,0,0,0.15);
+  }
+}
+
+[data-fs-product-card-image] {
+  height: 200px;
+  object-fit: cover;
+  transition: transform 0.3s ease;
+}
+
+[data-fs-product-card]:hover [data-fs-product-card-image] {
+  transform: scale(1.05);
+}
````
```suggestion
```scss
// Custom product card with hover effects
[data-fs-product-card] {
  transition: all 0.3s ease;
  border-radius: 12px;
  overflow: hidden;
  
  &:hover {
    transform: translateY(-4px);
    box-shadow: 0 8px 25px rgba(0,0,0,0.15);
    
    [data-fs-product-card-image] {
      transform: scale(1.05);
    }
  }
}

[data-fs-product-card-image] {
  height: 200px;
  object-fit: cover;
  transition: transform 0.3s ease;
}
```

renatamottam commented on docs/faststore/docs/styling/data-attributes.mdx:
```diff
@@ -0,0 +1,533 @@
+
+---
+
+## Best Practices
```
Why do we have two sections named "Best Practices"?

mariana-caetano replied:
I've combined the content into this section: https://github.com/vtexdocs/dev-portal-content/pull/2256/changes#diff-0ba3d7006eff78afb84baefbbb319c2aecc85ab6546902e4633efb1f86fa2dbaR390

renatamottam reviewed (commented):
Nice work, mari! Thanks for that 💛

mariana-caetano committed:
Apply suggestions from code review

Co-authored-by: Renata Motta <renata.motta@vtex.com>

mariana-caetano committed:
docs(data-attributes.mdx): update data attributes doc content according to peer review

Updated data attributes documentation to reflect changes in naming conventions, usage examples, and best practices. Enhanced clarity on component targeting and theming.

mariana-caetano committed:
Merge branch 'main' into data-attributes

mariana-caetano committed:
docs(data-attributes.mdx): organize section structure

mariana-caetano commented on docs/faststore/docs/styling/data-attributes.mdx:
```diff
@@ -0,0 +1,666 @@
+### Atoms
+
+<details>
+<summary><bold>Button</bold></summary>
```
```suggestion
<summary><b>Button</b></summary>
```

mariana-caetano committed:
Apply suggestions from code review

Co-authored-by: Mariana Caetano Pereira <67270558+mariana-caetano@users.noreply.github.com>

julia-rabello commented on docs/faststore/docs/styling/data-attributes.mdx:
padronizar a capitalização de `data attributes` / `Data Attributes` / `Data attributes` ao longo da doc

mariana-caetano replied:
@julia-rabello, pode revisar novamente? Padronizei a capitalização de `data attributes` e de alguns headings, troquei as tags `<details>` por tabelas e incluí  o link para o doc de design tokens na seção “Use design tokens”.

julia-rabello commented on docs/faststore/docs/styling/data-attributes.mdx:
````diff
@@ -0,0 +1,666 @@
+}
+```
+
+### Nested Elements
````
title case é o padrão em docs de FastStore? se não, sugiro usar sentence case em todos os headings

julia-rabello commented on docs/faststore/docs/styling/data-attributes.mdx:
```diff
@@ -0,0 +1,666 @@
+- `data-fs-navbar-links` - Navigation links
+- `data-fs-navbar-actions` - Action buttons area
+
+</details>
```
faria sentido simplificar isso com uma tabela?

mariana-caetano replied:
Faria sim! Vou modificar para uma tabela e adicionar links aos componentes citados.

julia-rabello commented on docs/faststore/docs/styling/data-attributes.mdx:
```diff
@@ -0,0 +1,666 @@
+
+### Use design tokens
+
+Override component styles using CSS custom properties (design tokens) rather than hardcoded values. This keeps styles consistent across the design system and makes future updates easier.
```
citar doc de design tokens novamente

mariana-caetano replied:
Boa!

mariana-caetano committed:
fix(data-attributes.mdx): review peer review suggestions

kaio-donadelli committed:
Review by the Localization team.

mariana-caetano committed:
Merge pull request #2797 from vtexdocs/l10n_data-attributes

LOC-25520: EDU-16348 Using data attributes

mariana-caetano committed:
Merge branch 'main' into data-attributes

mariana-caetano committed:
Merge branch 'main' into data-attributes
